### PR TITLE
Copy and display source location prominently on Kubernetes instances

### PR DESCRIPTION
**Pull request by maisem**
Reapplying #27840.

Following from #27830, this copies the source onto the instance and
displays the location of it prominently (keeping the download link for
anyone that just wants to curl it).

Example output (this tag doesn't exist yet):

---

Welcome to Kubernetes v1.4.0!

You can find documentation for Kubernetes at:
  http://docs.kubernetes.io/

The source for this release can be found at:
  /usr/local/share/doc/kubernetes/kubernetes-src.tar.gz
Or you can download it at:
  https://storage.googleapis.com/kubernetes-release/release/v1.4.0/kubernetes-src.tar.gz

It is based on the Kubernetes source at:
  https://github.com/kubernetes/kubernetes/tree/v1.4.0

For Kubernetes copyright and licensing information, see:
  /usr/local/share/doc/kubernetes/LICENSES

---

### CODE REVIEW

**Comment by andyzheng0831** [comment]
I am fine with the change. But as shown in https://github.com/kubernetes/kubernetes/issues/27764#issuecomment-228159892, I think it is better to ask @davidopp to add lgtm label

**Comment by alex-mohr** [comment]
Added retest-not-required because if this passes in PR builder, further testing doesn't help add confidence for this change and we can use the submit queue slot for something where it would.

**Comment by alex-mohr** [comment]
Per https://github.com/kubernetes/kubernetes/issues/27764#issuecomment-228196070 @maisem wrote:
"""
Upgrading the master worked.
cluster/gce/upgrade.sh -M v1.4.0-alpha.0.674+890ac5fdfea522

Upgrading the nodes also worked.
cluster/gce/upgrade.sh -N v1.4.0-alpha.0.674+890ac5fdfea522

So what @andyzheng0831 said is correct. There is an issue with v1.3.0-beta.2. If we upgrade to HEAD everything seems to be working fine.
"""

tl/dr: upgrading to v1.3.0-beta.2 is broken because it doesn't have the kubernetes-src.tar.gz file, but we expect to cut v.1.3.0 (final release) from HEAD and v1.4.0-alpha.0.674 is a reasonable proxy for that.

lgtm.

**Comment by alex-mohr** [comment]
FWIW, chatted with @spxtr and the "Jenkins GKE e2e" suite is not a merge bot blocker.  He's working on a fix, but that red X shouldn't stop this from merging.

**Comment by davidopp** [comment]
cc/ @gmarek @wojtek-t

File: cluster/gce/gci/configure.sh [hunk 1]
```diff
@@ -158,6 +158,9 @@ function install-kube-binary-config {
   fi
 
   cp "${KUBE_HOME}/kubernetes/LICENSES" "${KUBE_HOME}"
+  cp "${KUBE_HOME}/kubernetes/kubernetes-src.tar.gz" "${KUBE_HOME}"
```
Comment (yujuhong) [inline cluster/gce/gci/configure.sh]: Where does the script download `kubernetes-src.tar.gz`? This is not included in `kubernetes-release/release/v1.3.0-beta.2/kubernetes-server-linux-amd64.tar.gz `. My cluster upgrade failed because of this: `./cluster/gce/upgrade.sh -M v1.3.0-beta.2`
